exonum/src/runtime/rust/tests.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::proto::schema::tests::TimestampTx;
+
+use super::{
+    service::SystemService, ArtifactSpec, RustArtifactSpec, RustRuntime, TransactionContext,
+};
+use crate::crypto::{Hash, PublicKey};
+use crate::runtime::{
+    error::ExecutionError, CallInfo, DeployStatus, EnvContext, InstanceInitData, RuntimeEnvironment,
+};
+use crate::storage::{Database, Entry, MemoryDB};
+use protobuf::Message;
+
+service_interface! {
```

Comment:
I would prefer if possible a regular `trait` or `impl` blocks with the derive attribute than the IDE-unfriendly macroses.

Reply:
Totally agree.